oci-unit-tests/standalone-checks_test.sh

```diff
@@ -93,9 +105,119 @@ test_all_supported_architectures_are_available()
         done
     done
 
-    rm -rf "${utilsdir}"
-
     assertTrue "Not all supported architectures are available" "$ret"
 }
 
+test_images_hashes_are_equal()
+{
+    # We currently support only the "ubuntu" namespace, because it's
+    # the only one that is published on both registries.
+    local namespaces="ubuntu"
+    local images_list
+    local ret=0
+
+    if [ -z "${DOCKER_USERNAME}" ] || [ -z "${DOCKER_PASSWORD}" ]; then
+        echo "E: You must specify the DOCKER_USERNAME and DOCKER_PASSWORD environment variables." > /dev/stderr
+        exit 1
+    fi
+    export DOCKER_USERNAME DOCKER_REGISTRY
+
+    for namespace in ${namespaces}; do
+        # Querying the AWS registry is easier, and we assume that the
+        # list of images is the same for all registries.
+        images_list=$("${UTILSDIR}"/list-all-images.sh \
+                                   --registry "aws" \
+                                   --namespace "${namespace}")
+        if [ "${DOCKER_PACKAGE}" != "$(basename "$0" | sed 's@\(.*\)_test\.sh@\1@')" ]; then
+            # If the user has specified an image name to be tested, then
+            # we just test it.
+            if ! images_list=$(echo "${images_list}" | grep -Fw "${DOCKER_PACKAGE}"); then
+                continue
+            fi
+        fi
+        for image in ${images_list}; do
+            # Querying the AWS registry is easier, and we assume that the
+            # list of tags is the same for all registries.
```

Comment:
Should we always trust this assumption?
In the hypothetical scenario where we had just published a new image in a new namespace (or just a new tag), and the push operation failed for aws, this test could catch it in case we do not make this assumption. The  extra requests to dockerhub mighty be worth it to cover such corner cases.

I do understand this is not the purpose of this specific test though.

Reply:
I thought about this while writing this code.  I was also not happy to default to a specific registry for this query, but I thought that if we want to make sure that the same images are being published in all supported registries, then the best place to do this would be in another test (which I was planning to write).  But as I said I agree with your point, so I will see about making this part more robust.

Reply:
OK, I've pushed something that I think is better.  Please have a look :-).